Library/Formula/carina.rb

```diff
@@ -0,0 +1,30 @@
+class Carina < Formula
+  desc "Work with Swarm clusters on Carina"
+  homepage "https://github.com/getcarina/carina"
+  url "https://github.com/getcarina/carina.git",
+        :tag => "v0.7.1",
+        :revision => "885c4e298a9064e60a07dffd0143c38095def96e"
+  head "https://github.com/getcarina/carina.git"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+
+    carinapath = buildpath/"src/github.com/getcarina/carina"
+    carinapath.install Dir["{*,.git}"]
+
+    cd carinapath do
+      version = `git describe --tags 2> /dev/null`.strip
```

Comment:
Awesome, thanks! I dug too deep into `build` and other bits trying to get what was set for `tag` and `revision` with no luck.

Ended up making this much cleaner to have the commit detected in `carina`s Makefile as well.